src/components/application_manager/test/commands/mobile/system_request_test.cc

```diff
@@ -128,6 +140,133 @@ TEST_F(SystemRequestTest, Run_HTTP_FileName_no_binary_data_REJECTED) {
+       Run_RequestTypeAllowedAndRequestSubTypeAllowed_SendHMIRequest) {
+  MessageSharedPtr msg = CreateIVSUMessage();
+
+  (*msg)[am::strings::msg_params][am::strings::request_type] =
+      mobile_apis::RequestType::OEM_SPECIFIC;
+
+  const std::string request_subtype = "fakeSubType";
+  (*msg)[am::strings::msg_params][am::strings::request_subtype] =
+      request_subtype;
+
+  const std::vector<uint8_t> binary_data = {1u, 2u};
+  (*msg)[am::strings::params][am::strings::binary_data] = binary_data;
+
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId,
+                                   mobile_apis::RequestType::OEM_SPECIFIC))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestSubTypeAllowed(kAppPolicyId, request_subtype))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(app_mngr_,
+              SaveBinary(binary_data, kSystemFilesPath, kFileName, 0u))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  smart_objects::SmartObjectSPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
+  ASSERT_TRUE(command->Init());
+  command->Run();
+
+  EXPECT_EQ(MessageType::kRequest,
+            (*result)[am::strings::params][am::strings::message_type].asInt());
+  EXPECT_EQ(
+      mobile_apis::RequestType::OEM_SPECIFIC,
+      (*result)[am::strings::msg_params][am::strings::request_type].asInt());
+  EXPECT_EQ(
+      request_subtype,
+      (*msg)[am::strings::msg_params][am::strings::request_subtype].asString());
+}
+
+TEST_F(
+    SystemRequestTest,
+    Run_RequestTypeAllowedAndRequestSubTypeDisallowed_SendDisallowedResponse) {
+  PreConditions();
+  MockPolicySettings mock_policy_settings;
+  PolicyHandler policy_handler(mock_policy_settings, app_mngr_);
+  utils::SharedPtr<MockPolicyManager> mock_policy_manager_ptr =
+      utils::MakeShared<MockPolicyManager>();
+  policy_handler.SetPolicyManager(mock_policy_manager_ptr);
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(policy_handler));
+
+  // Request type allowed
+  EXPECT_CALL(*mock_policy_manager_ptr, GetAppRequestTypesState(kAppPolicyId))
+      .WillOnce(Return(policy::RequestType::State::AVAILABLE));
+
+  const std::vector<std::string> request_types = {"OEM_SPECIFIC"};
+  EXPECT_CALL(*mock_policy_manager_ptr, GetAppRequestTypes(kAppPolicyId))
+      .WillOnce(Return(request_types));
+
+  // Request subType Allowed
+  EXPECT_CALL(*mock_policy_manager_ptr,
+              GetAppRequestSubTypesState(kAppPolicyId))
+      .WillOnce(Return(policy::RequestSubType::State::AVAILABLE));
+  const std::vector<std::string> request_subtypes = {"fakeSubType"};
+  EXPECT_CALL(*mock_policy_manager_ptr, GetAppRequestSubTypes(kAppPolicyId))
+      .WillOnce(Return(request_subtypes));
+
+  MessageSharedPtr msg = CreateIVSUMessage();
```

Comment:
@BSolonenko Same as above

Reply:
@ValeriiMalkov I don't see in this any sense

Reply:
@BSolonenko Please, use C++11 https://adc.luxoft.com/confluence/pages/viewpage.action?pageId=307580753